TestTrigger_Error fails in nightly release

# Expected Behavior

Unit tests should consistently pass or fail; failures should consistently indicate real problems.

# Actual Behavior

When run as part of the nightly release, the test TestTrigger_Error fails with:

```
{"level":"error","ts":1602272424.723455,"caller":"cmd/root.go:217","msg":"Failed to resolve Trigger: failed to resolve bindings: error getting TriggerBinding : triggerbindings.triggers.tekton.dev \"git-event-binding\" is forbidden: User \"system:serviceaccount:default:default\" cannot get resource \"triggerbindings\" in API group \"triggers.tekton.dev\" in the namespace \"default\"","/triggers-eventid":"rnv4s","/trigger":"","stacktrace":"github.com/tektoncd/triggers/cmd/triggerrun/cmd.processTriggerSpec\n\t/workspace/src/github.com/tektoncd/triggers/cmd/triggerrun/cmd/root.go:217\ngithub.com/tektoncd/triggers/cmd/triggerrun/cmd.trigger\n\t/workspace/src/github.com/tektoncd/triggers/cmd/triggerrun/cmd/root.go:108\ngithub.com/tektoncd/triggers/cmd/triggerrun/cmd.TestTrigger_Error\n\t/workspace/src/github.com/tektoncd/triggers/cmd/triggerrun/cmd/root_test.go:45\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1127"}
    root_test.go:50: Error actual = fail to create resources: failed to resolve bindings: error getting TriggerBinding : triggerbindings.triggers.tekton.dev "git-event-binding" is forbidden: User "system:serviceaccount:default:default" cannot get resource "triggerbindings" in API group "triggers.tekton.dev" in the namespace "default", and Expected = fail to get clients: Failed to build config from the flags: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable.
--- FAIL: TestTrigger_Error (0.07s)
```

Failed twice using "latest" golang (15.2?):
* https://dashboard.dogfooding.tekton.dev/#/namespaces/default/pipelineruns/triggers-release-nightly-ngspw?pipelineTask=unit-tests&step=unit-test
* https://dashboard.dogfooding.tekton.dev/#/namespaces/default/pipelineruns/triggers-release-nightly-twkhk?pipelineTask=unit-tests&step=unit-test

Failed pinned to 14:
* https://dashboard.dogfooding.tekton.dev/#/namespaces/default/pipelineruns/triggers-release-nightly-bwbhq?pipelineTask=unit-tests&step=unit-test&view=details

# Steps to Reproduce the Problem

If you have permission you can manually trigger the nightly build with something like:

```
k --context dogfood create job --from cronjob/nightly-cron-trigger-triggers-nightly-release nightly-cron-trigger-triggers-nightly-release-manual-10092020-x
```

I couldn't reproduce it locally tho! At least it seems to be consistent for the nightly release.

![image](https://user-images.githubusercontent.com/432502/95626992-65199d80-0a49-11eb-8c58-d3f8359deb54.png)


# Additional Info

In #794 I remove the pinning of the golang version, and I believe the latest currently is 15.2, so it could be something to do with the tests failing only when run with that version, but when I ran them locally using that version (repeatedly! trying to use all the same flag!) I couldn't reproduce it with 14 or 15.

I also tried to pin to 14 and ran https://dashboard.dogfooding.tekton.dev/#/namespaces/default/pipelineruns/triggers-release-nightly-bwbhq?pipelineTask=unit-tests&step=unit-test&view=details to see if that would make a difference but it still failed.

--- a/cmd/triggerrun/cmd/root.go
+++ b/cmd/triggerrun/cmd/root.go
@@ -68,7 +68,7 @@ func init() {
 }
 
 func rootRun(cmd *cobra.Command, args []string) error {
-	err := trigger(triggerFile, httpPath, action, os.Stdout)
+	err := trigger(triggerFile, httpPath, action, kubeconfig, os.Stdout)
 	if err != nil {
 		return fmt.Errorf("fail to call trigger: %v", err)
 	}
@@ -76,7 +76,7 @@ func rootRun(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func trigger(triggerFile, httpPath, action string, writer io.Writer) error {
+func trigger(triggerFile, httpPath, action, kubeconfig string, writer io.Writer) error {
 	// Read HTTP request.
 	request, body, err := readHTTP(httpPath)
 	if err != nil {

--- a/cmd/triggerrun/cmd/root_test.go
+++ b/cmd/triggerrun/cmd/root_test.go
@@ -22,6 +22,7 @@ import (
 	"net/http"
 	"net/http/httputil"
 	"regexp"
+	"strings"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
@@ -42,18 +43,12 @@ import (
 func TestTrigger_Error(t *testing.T) {
 	//error case for show feature
 	buf := new(bytes.Buffer)
-	err := trigger("../testdata/trigger.yaml", "../testdata/http.txt", "show", buf)
+	err := trigger("../testdata/trigger.yaml", "../testdata/http.txt", "show", "BAD_KUBECONFIG", buf)
 
-	wantError := "fail to get clients: Failed to build config from the flags: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable"
+	wantError := "fail to get clients: Failed to build config from the flags:"
 
-	if err.Error() != wantError {
-		t.Errorf("Error actual = %v, and Expected = %v.", err, wantError)
-	}
-
-	gotValue := buf.Bytes()
-	wantValue := []uint8(nil)
-	if diff := cmp.Diff(wantValue, gotValue); diff != "" {
-		t.Errorf("Trigger mismatch -want +got: %s", diff)
+	if !strings.Contains(err.Error(), wantError) {
+		t.Errorf("Expected actual error to contain %s but got: %s", wantError, err.Error())
 	}
 }
 

